tests/utils/stl/util.py

```diff
@@ -0,0 +1,181 @@
+#===----------------------------------------------------------------------===##
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+#===----------------------------------------------------------------------===##
+
+from contextlib import contextmanager
+from pathlib import Path
+import os
+import platform
+import signal
+import subprocess
+import sys
+import tempfile
+import threading
+
+
+@contextmanager
+def guardedTempFilename(suffix='', prefix='', dir=None):
+    # Creates and yields a temporary filename within a with statement. The file
+    # is removed upon scope exit.
+    handle, name = tempfile.mkstemp(suffix=suffix, prefix=prefix, dir=dir)
+    os.close(handle)
+    yield name
+    Path(name).unlink(True)
+
+
+@contextmanager
+def guardedFilename(name):
+    # yields a filename within a with statement. The file is removed upon scope
+    # exit.
+    yield name
+    Path(name).unlink(True)
+
+
+@contextmanager
+def nullContext(value):
+    # yeilds a variable within a with statement. No action is taken upon scope

```

Comment:
Typo and capitalization:

```suggestion
    # Yields a variable within a with statement. No action is taken upon scope
```